launch/launch/logging/__init__.py

```diff
@@ -331,7 +333,12 @@ def get_screen_handler():
     See launch_config() documentation for screen logging configuration.
     """
     if launch_config.screen_handler is None:
-        launch_config.screen_handler = handlers.StreamHandler(sys.stdout)
+        stream = codecs.StreamWriter(sys.stdout, errors='replace')
+        stream.encode = lambda msg, errors='replace': (
+            msg.encode(locale.getpreferredencoding(False), errors).decode(
+                locale.getpreferredencoding(False), errors=errors),
```

Comment:
@dirk-thomas do we need to `decode()`? I think it'll be encoded again when it reaches `sys.stdout`.

Reply:
> do we need to decode()?

Yes, we need `decode()` since the API for writing only accepts `str` - not `bytes`.

> I think it'll be encoded again when it reaches `sys.stdout`.

Exactly that is why it needs to be decoded into a `str`.

Reply:
Hmm, I see the intent now. We _could_ use the binary raw stream e.g. `sys.stdout.buffer`, but if a client library replaces `sys.stdout` with another object it'll fall apart. I guess we'll have to wait for [`reconfigure()`](https://docs.python.org/3/library/io.html#io.TextIOWrapper.reconfigure).